pyproject.toml

```diff
@@ -2021,6 +2021,7 @@ module = [
     "tests.sentry.sentry_metrics.test_strings",
     "tests.sentry.services.test_http",
     "tests.sentry.services.test_hybrid_cloud",
+    "tests.sentry.services.test_organization_actions",
```

Comment:
new files should be typed -- this is a pretty easy one to get working as well:

```
tests/sentry/services/test_organization_actions.py:36: error: "OrganizationUpdateTest" has no attribute "create_organization"  [attr-defined]
tests/sentry/services/test_organization_actions.py:50: error: Cannot determine type of "slug"  [has-type]
tests/sentry/services/test_organization_actions.py:57: error: "OrganizationUpdateWithOutboxTest" has no attribute "create_organization"  [attr-defined]
tests/sentry/services/test_organization_actions.py:67: error: Cannot determine type of "slug"  [has-type]
tests/sentry/services/test_organization_actions.py:77: error: "OrganizationUpsertWithOutboxTest" has no attribute "create_organization"  [attr-defined]
tests/sentry/services/test_organization_actions.py:98: error: Cannot determine type of "slug"  [has-type]
tests/sentry/services/test_organization_actions.py:121: error: Cannot determine type of "slug"  [has-type]
tests/sentry/services/test_organization_actions.py:127: error: Cannot determine type of "slug"  [has-type]
```

Reply:
it looks like the attr-defined errors are due to the HC decorators being unsound.  the decorators should ~roughly have this signature:

```python
Func = TypeVar('Func', bound=Callable[..., Any])
@overload
def __call__(self, o: type[T]) -> type[T]: ...
@overload
def __call__(self, o: Func) -> Func: ...
```

Reply:
Do you happen to know which decorator(s) are responsible for this? I'm still fairly new to python/mypy but am happy to fix this if I know where to look 😅

Reply:
`region_silo_test` for example

Reply:
Hmm so I spent a while on Friday/this morning attempting to fix this in `silo.py` but ran into multiple issues attempting to overload the definition to fit all of the possible cases:
1. We take a function with the type `Callable[..., None]` (test method signature), and returns a wrapped test method with the same type definition
2. We take a `type[TestCase]` and return a `type[TestCase]` after wrapping its methods
3. We take a `None` and return a closure which takes a `type[TestCase]` and returns a `type[TestCase]` after wrapping its methods

I couldn't get a series of overload definitions that aligned properly for mypy to accept it and it's possible I'm just doing something incorrect with the way I'm defining overloads.

In the meantime I've accepted a suggestion from a teammate to simply subclass the TestCase using `BaseTestCase` which seems to alleviate the issue in the interim